lib/puppet_blacksmith/forge.rb

```diff
@@ -24,37 +28,60 @@ def initialize(username = nil, password = nil, url = nil)
         puts "Ignoring url entry in .puppetforge.yml: must point to the api server at #{PUPPETLABS_FORGE}, not the Forge webpage"
         self.url = PUPPETLABS_FORGE
       end
+      self.forge_type = forge_type unless forge_type.nil?
+      unless [FORGE_TYPE_PUPPET, FORGE_TYPE_ARTIFACTORY].include?(self.forge_type)
+        puts "Ignoring unknown forge type #{self.forge_type}"
+        self.forge_type = FORGE_TYPE_PUPPET
+      end
     end
 
-    def push!(name, package = nil)
+    def push!(author, name, package = nil)
```

Comment:
I think this API is also used externally. I'd prefer to add author as the last argument, defaulting to nil and then in have author default to username. That'd keep it API compatible while maintaining the flexibility.

Reply:
This was the one change I was a bit wary of, I've updated it to follow your suggestion.